packages/@sanity/desk-tool/src/DeskTool.tsx

```diff
@@ -57,25 +59,8 @@ export function DeskTool(props: DeskToolProps) {
   const {action, legacyEditDocumentId, type: schemaType, editDocumentId, params = {}} =
     routerState || {}
 
-  const paneKeys = useMemo(() => {
-    return routerPanes.reduce<string[]>(
-      (ids, group) => ids.concat(group.map((sibling) => sibling.id)),
-      ['root']
-    )
-  }, [routerPanes])
-
-  const paneGroups: RouterPaneGroup[] = useMemo(
-    () => ([[{id: 'root', params: {}}]] as RouterPaneGroup[]).concat(routerPanes || []),
-    [routerPanes]
-  )
-
-  const groupIndexes = useMemo(
-    () =>
-      routerPanes.reduce<number[]>(
-        (ids, group) => ids.concat(group.map((_, groupIndex) => groupIndex)),
-        []
-      ),
-    [routerPanes]
+  const panes = useUnique(
```

Comment:
Adding a comment 👍

Reply:
In this case, it’s because the `router.state.panes` is parsed from the `location.pathname` string, and will generate new objects for all segments, even when they didn’t change. Adding this prevents a few re-renders.

Reply:
i'm don't think `useUnique` is a good idea here. these objects are the resolved versions of the structure builder which are much bigger objects than the routerPanes.

routerPanes has `id`, `params`, and `payload` in a double array and that's it. the resolved panes include `id`, `title`, all options from any structure node, and all child resolver functions, and all menu item groups, and menu items (include their functions action handler function).

Reply:
Yes, totally true – my comment here was about the wrong thing (misread this for the router.state.panes, while this is actually the resolvedPanes, yeah). Removing `isUnique` here 👍